src/main/java/uk/gov/pay/api/model/links/PaymentLinks.java

```diff
@@ -12,6 +12,7 @@
 import static javax.ws.rs.HttpMethod.POST;
 
 @Schema(name = "PaymentLinks", description = "links for payment")
+@JsonInclude(value = JsonInclude.Include.NON_EMPTY)
```

Comment:
Might we have to be cautious with this, as it changes the API behaviour.

Currently we do include the capture and cancel links in search results if they are null
```
            "_links": {
                "self": {
                    "href": "https://publicapi.pymnt.uk/v1/payments/1jvgk16f8j1q8t1ec8pv9mnbbj",
                    "method": "GET"
                },
                "cancel": null,
                "events": {
                    "href": "https://publicapi.pymnt.uk/v1/payments/1jvgk16f8j1q8t1ec8pv9mnbbj/events",
                    "method": "GET"
                },
                "refunds": {
                    "href": "https://publicapi.pymnt.uk/v1/payments/1jvgk16f8j1q8t1ec8pv9mnbbj/refunds",
                    "method": "GET"
                },
                "capture": null
            },
```

But we don't include for get single payment:

```
    "_links": {
        "self": {
            "href": "https://publicapi.pymnt.uk/v1/payments/b930o02iq7nfdvfnk7opnkofsj",
            "method": "GET"
        },
        "events": {
            "href": "https://publicapi.pymnt.uk/v1/payments/b930o02iq7nfdvfnk7opnkofsj/events",
            "method": "GET"
        },
        "refunds": {
            "href": "https://publicapi.pymnt.uk/v1/payments/b930o02iq7nfdvfnk7opnkofsj/refunds",
            "method": "GET"
        }
    }
```

I'm not quite sure the mechanism for excluding the null links for GET single payment responses if this annotation wasn't present before, but is it safest to ensure the behaviour is unchanged?

Reply:
> Currently we do include the capture and cancel links in search results if they are null

I feel like this wouldn't be a problem. If they can be null, any client would have to test that they are there to begin with.

Reply:
Despite all the above, I'm happy to revert the deletion of `PaymentLinksForSearch` if you think that's safer!

Reply:
I think we've had concerns in the past about making similar changes. I think there is a possibility for consumers to be doing interesting things that might break if we stop including properties when they're null when we previously including them. It possibly constitutes a breaking change to the API.
If reverting the deletion of `PaymentLinksForSearch` still allows you to improve things in some way, maybe that would be easiest.

Reply:
Happy to do that, will revert.

Reply:
Ok, [done](https://github.com/alphagov/pay-publicapi/pull/1781/commits/9274f5775bc146d1ba73380aed0e0d2702c9afda).